File: src/main/java/com/bot/commands/voice/PauseCommand.java
package com.bot.commands.voice;

import com.bot.commands.VoiceCommand;
import com.bot.voice.VoiceSendHandler;
import com.jagrosh.jdautilities.command.CommandEvent;

public class PauseCommand extends VoiceCommand {

	public PauseCommand() {
		this.name = "pause";
		this.arguments = "";
		this.help = "Pauses or resumes the stream";
	}

	@Override
	protected void executeCommand(CommandEvent commandEvent) {
		metricsManager.markCommand(this, commandEvent.getAuthor(), commandEvent.getGuild());

		VoiceSendHandler handler = (VoiceSendHandler) commandEvent.getGuild().getAudioManager().getSendingHandler();
		if (handler == null) {
			commandEvent.reply(commandEvent.getClient().getWarning() + " I am not connected to a voice channel.");
		}
		else {
			if (handler.getPlayer().isPaused()) {
				handler.getPlayer().setPaused(false);
				commandEvent.reply(commandEvent.getClient().getSuccess() + " Resumed stream.");
			}
			else {
				handler.getPlayer().setPaused(true);
				commandEvent.reply(commandEvent.getClient().getSuccess() + " Paused stream.");
			}
		}
	}
}

File: src/main/java/com/bot/commands/voice/PauseCommand.kt
package com.bot.commands.voice

import com.bot.commands.VoiceCommand
import com.jagrosh.jdautilities.command.CommandEvent
import datadog.trace.api.Trace

class PauseCommand : VoiceCommand() {
    init {
        name = "pause"
        aliases = arrayOf("resume")
        arguments = ""
        help = "Pauses or resumes the stream"
    }

    @Trace(operationName = "executeCommand", resourceName = "pause")
    override fun executeCommand(commandEvent: CommandEvent) {
        val guildVoiceConnection = guildVoiceProvider.getGuildVoiceConnection(commandEvent.guild)
        if (!guildVoiceConnection.isConnected()) {
            commandEvent.replyWarning(" I am not currently connected to voice.")
            return
        }
        val newState = !guildVoiceConnection.getPaused()
        guildVoiceConnection.setPaused(newState)
        commandEvent.replySuccess("Stream has been ${if (newState) "paused" else "resumed"}")
    }
}